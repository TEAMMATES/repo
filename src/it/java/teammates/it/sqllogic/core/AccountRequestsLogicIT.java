package teammates.it.sqllogic.core;

import java.time.Instant;
import java.util.UUID;

import org.testng.annotations.Test;

import teammates.common.datatransfer.AccountRequestStatus;
import teammates.common.exception.EntityAlreadyExistsException;
import teammates.common.exception.EntityDoesNotExistException;
import teammates.common.exception.InvalidParametersException;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.AccountRequestsLogic;
import teammates.storage.sqlapi.AccountRequestsDb;
import teammates.storage.sqlentity.AccountRequest;

/**
 * SUT: {@link AccountRequestsLogic}.
 */
public class AccountRequestsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private AccountRequestsLogic accountRequestsLogic = AccountRequestsLogic.inst();

    @Test
    public void testGetAccountRequest_nonExistentAccountRequest_returnsNull() {
        UUID id = UUID.randomUUID();
        AccountRequest actualAccountRequest = accountRequestsLogic.getAccountRequest(id);
        assertNull(actualAccountRequest);
    }

    @Test
    public void testGetAccountRequest_existingAccountRequest_getsSuccessfully() throws InvalidParametersException {
        AccountRequest expectedAccountRequest =
                new AccountRequest("test@gmail.com", "name", "institute", AccountRequestStatus.PENDING, "comments");
        UUID id = expectedAccountRequest.getId();
        accountRequestsLogic.createAccountRequest(expectedAccountRequest);
        AccountRequest actualAccountRequest = accountRequestsLogic.getAccountRequest(id);
        assertEquals(expectedAccountRequest, actualAccountRequest);
    }

    @Test
    public void testResetAccountRequest()
            throws EntityAlreadyExistsException, InvalidParametersException, EntityDoesNotExistException {

        ______TS("success: create account request and update registeredAt field");

        String name = "name lee";
        String email = "email@gmail.com";
        String institute = "institute";
        AccountRequestStatus status = AccountRequestStatus.PENDING;
        String comments = "comments";

        AccountRequest toReset = accountRequestsLogic.createAccountRequest(name, email, institute, status, comments);
        AccountRequestsDb accountRequestsDb = AccountRequestsDb.inst();

        toReset.setRegisteredAt(Instant.now());
        toReset = accountRequestsDb.getAccountRequest(email, institute);

        assertNotNull(toReset);
        assertNotNull(toReset.getRegisteredAt());

        ______TS("success: reset account request that already exists");

        AccountRequest resetted = accountRequestsLogic.resetAccountRequest(email, institute);

        assertNull(resetted.getRegisteredAt());

        ______TS("success: test delete account request");

        accountRequestsLogic.deleteAccountRequest(email, institute);

        assertNull(accountRequestsLogic.getAccountRequest(email, institute));

        ______TS("failure: reset account request that does not exist");

        assertThrows(EntityDoesNotExistException.class,
                () -> accountRequestsLogic.resetAccountRequest(name, institute));
    }
}
