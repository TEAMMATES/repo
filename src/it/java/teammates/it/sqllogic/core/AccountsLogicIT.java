package teammates.it.sqllogic.core;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

import org.testng.annotations.Test;

import teammates.common.datatransfer.NotificationStyle;
import teammates.common.datatransfer.NotificationTargetUser;
import teammates.common.exception.EntityAlreadyExistsException;
import teammates.common.exception.EntityDoesNotExistException;
import teammates.common.exception.InvalidParametersException;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.AccountsLogic;
import teammates.sqllogic.core.NotificationsLogic;
import teammates.storage.sqlapi.AccountsDb;
import teammates.storage.sqlentity.Account;
import teammates.storage.sqlentity.Notification;
import teammates.storage.sqlentity.ReadNotification;

/**
 * SUT: {@link AccountsLogic}.
 */
public class AccountsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private AccountsLogic accountsLogic = AccountsLogic.inst();
    private NotificationsLogic notificationsLogic = NotificationsLogic.inst();

    private AccountsDb accountsDb = AccountsDb.inst();

    @Test
    public void testUpdateReadNotifications()
            throws EntityAlreadyExistsException, InvalidParametersException, EntityDoesNotExistException {
        ______TS("success: mark notification as read");
        Account account = new Account("google-id", "name", "email@teammates.com");
        Notification notification = new Notification(Instant.parse("2011-01-01T00:00:00Z"),
                Instant.parse("2099-01-01T00:00:00Z"), NotificationStyle.DANGER, NotificationTargetUser.GENERAL,
                "A deprecation note", "<p>Deprecation happens in three minutes</p>");
        accountsDb.createAccount(account);
        notificationsLogic.createNotification(notification);

        String googleId = account.getGoogleId();
        UUID notificationId = notification.getId();
        accountsLogic.updateReadNotifications(googleId, notificationId, notification.getEndTime());

        Account actualAccount = accountsDb.getAccountByGoogleId(googleId);
        List<ReadNotification> accountReadNotifications = actualAccount.getReadNotifications();
        assertEquals(1, accountReadNotifications.size());
        assertSame(actualAccount, accountReadNotifications.get(0).getAccount());
        assertSame(notification, accountReadNotifications.get(0).getNotification());
    }
}
