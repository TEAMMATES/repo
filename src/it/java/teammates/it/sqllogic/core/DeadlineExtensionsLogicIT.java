package teammates.it.sqllogic.core;

import java.time.Instant;

import org.testng.annotations.BeforeClass;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import teammates.common.datatransfer.SqlDataBundle;
import teammates.common.exception.EntityDoesNotExistException;
import teammates.common.exception.InvalidParametersException;
import teammates.common.util.HibernateUtil;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.DeadlineExtensionsLogic;
import teammates.storage.sqlentity.DeadlineExtension;
import teammates.storage.sqlentity.FeedbackSession;
import teammates.storage.sqlentity.Student;

/**
 * SUT: {@link DeadlineExtensionsLogic}.
 */
public class DeadlineExtensionsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private DeadlineExtensionsLogic deadlineExtensionsLogic = DeadlineExtensionsLogic.inst();
    private SqlDataBundle typicalDataBundle;

    @Override
    @BeforeClass
    public void setupClass() {
        super.setupClass();
        typicalDataBundle = getTypicalSqlDataBundle();
    }

    @Override
    @BeforeMethod
    protected void setUp() throws Exception {
        super.setUp();
        persistDataBundle(typicalDataBundle);
        HibernateUtil.flushSession();
        HibernateUtil.clearSession();
    }

    @Test
    public void testGetExtendedDeadline_extensionExists_success() {
        FeedbackSession feedbackSession = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        Student student = typicalDataBundle.students.get("student1InCourse1");

        assert student != null;
        Instant extendedDeadlineForStudent = deadlineExtensionsLogic.getExtendedDeadlineForUser(feedbackSession, student);

        assertNotNull(extendedDeadlineForStudent);
        assertEquals(Instant.parse("2027-04-30T23:00:00Z"), extendedDeadlineForStudent);
    }

    @Test
    public void testGetExtendedDeadline_extensionDoesNotExist_null() {
        FeedbackSession feedbackSession = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        Student student = typicalDataBundle.students.get("student2InCourse1");
        Instant extendedDeadlineForStudent = deadlineExtensionsLogic.getExtendedDeadlineForUser(feedbackSession, student);

        assertNull(extendedDeadlineForStudent);
    }

    @Test
    public void testUpdateDeadlineExtension()
            throws EntityDoesNotExistException, InvalidParametersException {
        FeedbackSession feedbackSession = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        Student student = typicalDataBundle.students.get("student1InCourse1");

        DeadlineExtension de = typicalDataBundle.deadlineExtensions.get("student1InCourse1Session1");

        Instant newEndTime = Instant.parse("2028-04-30T23:00:00Z");
        de.setEndTime(newEndTime);
        deadlineExtensionsLogic.updateDeadlineExtension(de);

        assertEquals(newEndTime, deadlineExtensionsLogic.getExtendedDeadlineForUser(feedbackSession, student));
    }

    @Test
    public void testUpdateDeadlineExtension_invalidParameters_originalUnchanged() {
        FeedbackSession feedbackSession = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        Student student = typicalDataBundle.students.get("student1InCourse1");

        DeadlineExtension de = typicalDataBundle.deadlineExtensions.get("student1InCourse1Session1");
        Instant originalEndTime = de.getEndTime();

        Instant invalidEndTime = Instant.parse("2011-04-01T22:00:00Z");
        assert invalidEndTime.isBefore(feedbackSession.getStartTime());
        de.setEndTime(invalidEndTime);

        assertThrows(InvalidParametersException.class, () -> deadlineExtensionsLogic.updateDeadlineExtension(de));

        assertEquals(originalEndTime, deadlineExtensionsLogic.getExtendedDeadlineForUser(feedbackSession, student));
    }

}
