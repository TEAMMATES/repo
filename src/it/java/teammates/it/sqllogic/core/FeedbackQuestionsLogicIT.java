package teammates.it.sqllogic.core;

import java.util.ArrayList;
import java.util.List;

import org.testng.annotations.BeforeClass;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import teammates.common.datatransfer.FeedbackParticipantType;
import teammates.common.datatransfer.SqlDataBundle;
import teammates.common.datatransfer.questions.FeedbackTextQuestionDetails;
import teammates.common.exception.InvalidParametersException;
import teammates.common.util.HibernateUtil;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.FeedbackQuestionsLogic;
import teammates.storage.sqlentity.FeedbackQuestion;
import teammates.storage.sqlentity.FeedbackSession;

/**
 * SUT: {@link FeedbackQuestionsLogic}.
 */
public class FeedbackQuestionsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private FeedbackQuestionsLogic fqLogic = FeedbackQuestionsLogic.inst();

    private SqlDataBundle typicalDataBundle;

    @Override
    @BeforeClass
    public void setupClass() {
        super.setupClass();
        typicalDataBundle = getTypicalSqlDataBundle();
    }

    @Override
    @BeforeMethod
    protected void setUp() throws Exception {
        super.setUp();
        persistDataBundle(typicalDataBundle);
        HibernateUtil.flushSession();
    }

    @Test
    public void testCreateFeedbackQuestion() throws InvalidParametersException {
        FeedbackSession fs = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        FeedbackTextQuestionDetails newQuestionDetails = new FeedbackTextQuestionDetails("New question text.");
        List<FeedbackParticipantType> showTos = new ArrayList<>();
        showTos.add(FeedbackParticipantType.INSTRUCTORS);
        FeedbackQuestion newQuestion = FeedbackQuestion.makeQuestion(fs, 6, "This is a new text question",
                FeedbackParticipantType.STUDENTS, FeedbackParticipantType.OWN_TEAM_MEMBERS, -100,
                showTos, showTos, showTos, newQuestionDetails);

        newQuestion = fqLogic.createFeedbackQuestion(newQuestion);

        FeedbackQuestion actualQuestion = fqLogic.getFeedbackQuestion(newQuestion.getId());

        verifyEquals(newQuestion, actualQuestion);
    }

    @Test
    public void testGetFeedbackQuestionsForSession() {
        FeedbackSession fs = typicalDataBundle.feedbackSessions.get("session1InCourse1");
        FeedbackQuestion fq1 = typicalDataBundle.feedbackQuestions.get("qn1InSession1InCourse1");
        FeedbackQuestion fq2 = typicalDataBundle.feedbackQuestions.get("qn2InSession1InCourse1");
        FeedbackQuestion fq3 = typicalDataBundle.feedbackQuestions.get("qn3InSession1InCourse1");
        FeedbackQuestion fq4 = typicalDataBundle.feedbackQuestions.get("qn4InSession1InCourse1");
        FeedbackQuestion fq5 = typicalDataBundle.feedbackQuestions.get("qn5InSession1InCourse1");
        FeedbackQuestion fq6 = typicalDataBundle.feedbackQuestions.get("qn6InSession1InCourse1NoResponses");

        List<FeedbackQuestion> expectedQuestions = List.of(fq1, fq2, fq3, fq4, fq5, fq6);

        List<FeedbackQuestion> actualQuestions = fqLogic.getFeedbackQuestionsForSession(fs);

        assertEquals(expectedQuestions.size(), actualQuestions.size());
        assertTrue(expectedQuestions.containsAll(actualQuestions));
    }

}
