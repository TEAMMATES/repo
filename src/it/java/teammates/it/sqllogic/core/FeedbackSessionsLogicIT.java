package teammates.it.sqllogic.core;

import org.testng.annotations.BeforeClass;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import teammates.common.datatransfer.SqlDataBundle;
import teammates.common.exception.EntityDoesNotExistException;
import teammates.common.exception.InvalidParametersException;
import teammates.common.util.HibernateUtil;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.FeedbackSessionsLogic;
import teammates.storage.sqlentity.FeedbackSession;

/**
 * SUT: {@link FeedbackSessionsLogic}.
 */
public class FeedbackSessionsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private FeedbackSessionsLogic fsLogic = FeedbackSessionsLogic.inst();

    private SqlDataBundle typicalDataBundle;

    @Override
    @BeforeClass
    public void setupClass() {
        super.setupClass();
        typicalDataBundle = getTypicalSqlDataBundle();
    }

    @Override
    @BeforeMethod
    protected void setUp() throws Exception {
        super.setUp();
        persistDataBundle(typicalDataBundle);
        HibernateUtil.flushSession();
    }

    @Test
    public void testPublishFeedbackSession()
            throws InvalidParametersException, EntityDoesNotExistException {
        FeedbackSession unpublishedFs = typicalDataBundle.feedbackSessions.get("unpublishedSession1InTypicalCourse");

        FeedbackSession publishedFs1 = fsLogic.publishFeedbackSession(
                unpublishedFs.getName(), unpublishedFs.getCourse().getId());

        assertEquals(publishedFs1.getName(), unpublishedFs.getName());
        assertTrue(publishedFs1.isPublished());

        assertThrows(InvalidParametersException.class, () -> fsLogic.publishFeedbackSession(
                publishedFs1.getName(), publishedFs1.getCourse().getId()));
        assertThrows(EntityDoesNotExistException.class, () -> fsLogic.publishFeedbackSession(
                "non-existent name", unpublishedFs.getCourse().getId()));
        assertThrows(EntityDoesNotExistException.class, () -> fsLogic.publishFeedbackSession(
                unpublishedFs.getName(), "random-course-id"));
    }

    @Test
    public void testUnpublishFeedbackSession()
            throws InvalidParametersException, EntityDoesNotExistException {
        FeedbackSession publishedFs = typicalDataBundle.feedbackSessions.get("session1InCourse1");

        FeedbackSession unpublishedFs1 = fsLogic.unpublishFeedbackSession(
                publishedFs.getName(), publishedFs.getCourse().getId());

        assertEquals(unpublishedFs1.getName(), publishedFs.getName());
        assertFalse(unpublishedFs1.isPublished());

        assertThrows(InvalidParametersException.class, () -> fsLogic.unpublishFeedbackSession(
                unpublishedFs1.getName(), unpublishedFs1.getCourse().getId()));
        assertThrows(EntityDoesNotExistException.class, () -> fsLogic.unpublishFeedbackSession(
                "non-existent name", publishedFs.getCourse().getId()));
        assertThrows(EntityDoesNotExistException.class, () -> fsLogic.unpublishFeedbackSession(
                publishedFs.getName(), "random-course-id"));
    }

}
