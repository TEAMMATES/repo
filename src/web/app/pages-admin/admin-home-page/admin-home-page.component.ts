import { Component } from '@angular/core';
import { concat, Observable, of } from 'rxjs';
import { catchError, tap } from 'rxjs/operators';
import { AccountService } from '../../../services/account.service';
import { JoinLink } from '../../../types/api-output';
import { ErrorMessageOutput } from '../../error-message-output';

interface InstructorData {
  name: string;
  email: string;
  institution: string;
  status: string;
  joinLink?: string;
  message?: string;
}

/**
 * Admin home page.
 */
@Component({
  selector: 'tm-admin-home-page',
  templateUrl: './admin-home-page.component.html',
  styleUrls: ['./admin-home-page.component.scss'],
})
export class AdminHomePageComponent {

  instructorDetails: string = '';
  instructorName: string = '';
  instructorEmail: string = '';
  instructorInstitution: string = '';

  instructorsConsolidated: InstructorData[] = [];
  activeRequests: number = 0;

  isAddingInstructors: boolean = false;

  constructor(private accountService: AccountService) {}

  /**
   * Validates and adds the instructor details filled with first form.
   */
  validateAndAddInstructorDetails(): void {
    const invalidLines: string[] = [];
    for (const instructorDetail of this.instructorDetails.split(/\r?\n/)) {
      const instructorDetailSplit: string[] = instructorDetail.split(/[|\t]/).map((item: string) => item.trim());
      if (instructorDetailSplit.length < 3) {
        // TODO handle should add all instructors when promptederror
        invalidLines.push(instructorDetail);
        continue;
      }
      if (!instructorDetailSplit[0] || !instructorDetailSplit[1] || !instructorDetailSplit[2]) {
        // TODO handle error
        invalidLines.push(instructorDetail);
        continue;
      }
      this.instructorsConsolidated.push({
        name: instructorDetailSplit[0],
        email: instructorDetailSplit[1],
        institution: instructorDetailSplit[2],
        status: 'PENDING',
      });
    }
    this.instructorDetails = invalidLines.join('\r\n');
  }

  /**
   * Validates and adds the instructor detail filled with second form.
   */
  validateAndAddInstructorDetail(): void {
    if (!this.instructorName || !this.instructorEmail || !this.instructorInstitution) {
      // TODO handle error
      return;
    }
    this.instructorsConsolidated.push({
      name: this.instructorName,
      email: this.instructorEmail,
      institution: this.instructorInstitution,
      status: 'PENDING',
    });
    this.instructorName = '';
    this.instructorEmail = '';
    this.instructorInstitution = '';
  }

  /**
   * Adds the instructor at the i-th index.
   */
  addInstructor(i: number): void {
    this.asyncAddInstructor(i)
      ?.subscribe({ complete: () => this.isAddingInstructors = false });
  }

  /**
   * Adds the instructor at the i-th index.
   */
  asyncAddInstructor(i: number): Observable<JoinLink> | null {
    const instructor: InstructorData = this.instructorsConsolidated[i];
    if (instructor.status !== 'PENDING' && instructor.status !== 'FAIL') {
      return null;
    }
    this.activeRequests += 1;
    instructor.status = 'ADDING';

    this.isAddingInstructors = true;
    return this.accountService.createAccount({
      instructorEmail: instructor.email,
      instructorName: instructor.name,
      instructorInstitution: instructor.institution,
    }).pipe(
      tap((resp: JoinLink) => {
        instructor.status = 'SUCCESS';
        instructor.joinLink = resp.joinLink;
        this.activeRequests -= 1;
      }, (resp: ErrorMessageOutput) => {
        instructor.status = 'FAIL';
        instructor.message = resp.error.message;
        this.activeRequests -= 1;
      }),
      // Lift error handling to tap operator and catch the final error
      catchError((_err: ErrorMessageOutput, _caught: Observable<JoinLink>) => of({ joinLink: '' })),
    );
  }

  /**
   * Cancels the instructor at the i-th index.
   */
  cancelInstructor(i: number): void {
    this.instructorsConsolidated.splice(i, 1);
  }

  /**
   * Adds all the pending and failed-to-add instructors one by one.
   * Refer to https://github.com/TEAMMATES/teammates/issues/11039
   */
  addAllInstructors(): void {
    concat(
      ...this.instructorsConsolidated
        .map((_instr: InstructorData, index: number): Observable<JoinLink> | null => this.asyncAddInstructor(index))
        .filter((req: Observable<JoinLink> | null): boolean => !!req)
        .map((req: Observable<JoinLink> | null): Observable<JoinLink> => req as Observable<JoinLink>),
    ).subscribe({ complete: () => this.isAddingInstructors = false });
  }

}
