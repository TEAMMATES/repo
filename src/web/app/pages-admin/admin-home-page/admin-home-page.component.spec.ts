import { HttpClientTestingModule } from '@angular/common/http/testing';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { AdminHomePageComponent } from './admin-home-page.component';

describe('AdminHomePageComponent', () => {
  let component: AdminHomePageComponent;
  let fixture: ComponentFixture<AdminHomePageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [AdminHomePageComponent],
      imports: [
        FormsModule,
        HttpClientTestingModule,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AdminHomePageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default view', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with some instructors details', () => {
    component.instructorsConsolidated = [
      {
        name: 'Instructor A',
        email: 'instructora@example.com',
        institution: 'Sample Institution A',
        status: 'PENDING',
        joinLink: 'This should not be displayed',
        message: 'This should not be displayed',
      },
      {
        name: 'Instructor B',
        email: 'instructorb@example.com',
        institution: 'Sample Institution B',
        status: 'SUCCESS',
        joinLink: 'http://localhost:4200/web/join',
        message: 'This should not be displayed',
      },
      {
        name: 'Instructor C',
        email: 'instructorc@example.com',
        institution: 'Sample Institution C',
        status: 'FAIL',
        joinLink: 'This should not be displayed',
        message: 'The instructor cannot be added for some reason',
      },
    ];
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with disabled adding instructor button if there are active requests', () => {
    component.instructorsConsolidated = [
      {
        name: 'Instructor A',
        email: 'instructora@example.com',
        institution: 'Sample Institution A',
        status: 'ADDING',
        joinLink: 'This should not be displayed',
        message: 'This should not be displayed',
      },
      {
        name: 'Instructor B',
        email: 'instructorb@example.com',
        institution: 'Sample Institution B',
        status: 'PENDING',
        joinLink: 'This should not be displayed',
        message: 'This should not be displayed',
      },
    ];
    component.activeRequests = 1;

    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

});
