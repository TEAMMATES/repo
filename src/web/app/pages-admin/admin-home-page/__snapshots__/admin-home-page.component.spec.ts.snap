// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminHomePageComponent should snap with default view 1`] = `
<tm-admin-home-page
  accountService={[Function AccountService]}
  activeRequests="0"
  instructorDetails=""
  instructorEmail=""
  instructorInstitution=""
  instructorName=""
  instructorsConsolidated={[Function Array]}
>
  <div
    class="card bg-light top-padded"
  >
    <div
      class="card-body"
    >
      <div>
        <strong>
          Adding Multiple Instructors
        </strong>
        <p
          class="text-muted"
        >
          <i
            class="fas fa-exclamation-circle"
          />
           Add Instructor Details in the format: Name | Email | Institution 
        </p>
        <textarea
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
        />
      </div>
      <div
        class="top-padded"
      >
        <button
          class="btn btn-primary"
        >
          Add Instructors
        </button>
      </div>
    </div>
  </div><div
    class="card bg-light top-padded"
  >
    <div
      class="card-body"
    >
      <div>
        <strong>
          Adding a Single Instructor
        </strong>
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Name:
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Email: 
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Institution: 
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <button
          class="btn btn-primary"
        >
          Add Instructor
        </button>
      </div>
    </div>
  </div>
</tm-admin-home-page>
`;

exports[`AdminHomePageComponent should snap with disabled adding instructor button if there are active requests 1`] = `
<tm-admin-home-page
  accountService={[Function AccountService]}
  activeRequests={[Function Number]}
  instructorDetails=""
  instructorEmail=""
  instructorInstitution=""
  instructorName=""
  instructorsConsolidated={[Function Array]}
>
  <div
    class="card bg-light top-padded"
  >
    <div
      class="card-body"
    >
      <div>
        <strong>
          Adding Multiple Instructors
        </strong>
        <p
          class="text-muted"
        >
          <i
            class="fas fa-exclamation-circle"
          />
           Add Instructor Details in the format: Name | Email | Institution 
        </p>
        <textarea
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
        />
      </div>
      <div
        class="top-padded"
      >
        <button
          class="btn btn-primary"
        >
          Add Instructors
        </button>
      </div>
    </div>
  </div><div
    class="card bg-light top-padded"
  >
    <div
      class="card-body"
    >
      <div>
        <strong>
          Adding a Single Instructor
        </strong>
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Name:
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Email: 
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Institution: 
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <button
          class="btn btn-primary"
        >
          Add Instructor
        </button>
      </div>
    </div>
  </div><div
    class="card bg-light top-padded"
  >
    <div
      class="card-header bg-primary text-white"
    >
      <strong>
        Result
      </strong>
    </div>
    <div
      class="card-body"
    >
      <div
        class="table-responsive"
      >
        <table
          class="table table-striped table-hover"
        >
          <thead>
            <tr>
              <th>
                Name
              </th>
              <th>
                Email
              </th>
              <th>
                Institution
              </th>
              <th>
                Action
              </th>
              <th>
                Status
              </th>
              <th>
                Message
              </th>
            </tr>
          </thead>
          <tbody>
            
            <tr>
              <td>
                Instructor A
              </td>
              <td>
                instructora@example.com
              </td>
              <td>
                Sample Institution A
              </td>
              <td>
                
              </td>
              <td>
                ADDING
              </td>
              <td
                style="width: 500px;"
              >
                
                
              </td>
            </tr>
            <tr>
              <td>
                Instructor B
              </td>
              <td>
                instructorb@example.com
              </td>
              <td>
                Sample Institution B
              </td>
              <td>
                
                <div>
                  <button
                    class="btn btn-sm btn-primary"
                    disabled=""
                    style="margin-right: 10px;"
                  >
                    Add Instructor
                  </button>
                  <button
                    class="btn btn-sm btn-danger"
                  >
                    Cancel
                  </button>
                </div>
              </td>
              <td>
                PENDING
              </td>
              <td
                style="width: 500px;"
              >
                
                
              </td>
            </tr>
          </tbody>
        </table>
        <button
          class="btn btn-primary"
          disabled=""
        >
          Add All Instructors
        </button>
      </div>
    </div>
  </div>
</tm-admin-home-page>
`;

exports[`AdminHomePageComponent should snap with some instructors details 1`] = `
<tm-admin-home-page
  accountService={[Function AccountService]}
  activeRequests="0"
  instructorDetails=""
  instructorEmail=""
  instructorInstitution=""
  instructorName=""
  instructorsConsolidated={[Function Array]}
>
  <div
    class="card bg-light top-padded"
  >
    <div
      class="card-body"
    >
      <div>
        <strong>
          Adding Multiple Instructors
        </strong>
        <p
          class="text-muted"
        >
          <i
            class="fas fa-exclamation-circle"
          />
           Add Instructor Details in the format: Name | Email | Institution 
        </p>
        <textarea
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
        />
      </div>
      <div
        class="top-padded"
      >
        <button
          class="btn btn-primary"
        >
          Add Instructors
        </button>
      </div>
    </div>
  </div><div
    class="card bg-light top-padded"
  >
    <div
      class="card-body"
    >
      <div>
        <strong>
          Adding a Single Instructor
        </strong>
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Name:
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Email: 
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <strong>
          Institution: 
        </strong>
        <input
          class="form-control ng-untouched ng-pristine ng-valid"
          ng-reflect-model=""
          type="text"
        />
      </div>
      <div
        class="top-padded"
      >
        <button
          class="btn btn-primary"
        >
          Add Instructor
        </button>
      </div>
    </div>
  </div><div
    class="card bg-light top-padded"
  >
    <div
      class="card-header bg-primary text-white"
    >
      <strong>
        Result
      </strong>
    </div>
    <div
      class="card-body"
    >
      <div
        class="table-responsive"
      >
        <table
          class="table table-striped table-hover"
        >
          <thead>
            <tr>
              <th>
                Name
              </th>
              <th>
                Email
              </th>
              <th>
                Institution
              </th>
              <th>
                Action
              </th>
              <th>
                Status
              </th>
              <th>
                Message
              </th>
            </tr>
          </thead>
          <tbody>
            
            <tr>
              <td>
                Instructor A
              </td>
              <td>
                instructora@example.com
              </td>
              <td>
                Sample Institution A
              </td>
              <td>
                
                <div>
                  <button
                    class="btn btn-sm btn-primary"
                    style="margin-right: 10px;"
                  >
                    Add Instructor
                  </button>
                  <button
                    class="btn btn-sm btn-danger"
                  >
                    Cancel
                  </button>
                </div>
              </td>
              <td>
                PENDING
              </td>
              <td
                style="width: 500px;"
              >
                
                
              </td>
            </tr>
            <tr>
              <td>
                Instructor B
              </td>
              <td>
                instructorb@example.com
              </td>
              <td>
                Sample Institution B
              </td>
              <td>
                
              </td>
              <td>
                SUCCESS
              </td>
              <td
                style="width: 500px;"
              >
                
                <span>
                   Instructor "Instructor B" has been successfully created. [
                  <a
                    href="http://localhost:4200/web/join"
                  >
                    join link
                  </a>
                  ] 
                </span>
                
              </td>
            </tr>
            <tr>
              <td>
                Instructor C
              </td>
              <td>
                instructorc@example.com
              </td>
              <td>
                Sample Institution C
              </td>
              <td>
                
                <div>
                  <button
                    class="btn btn-sm btn-primary"
                    style="margin-right: 10px;"
                  >
                    Add Instructor
                  </button>
                  <button
                    class="btn btn-sm btn-danger"
                  >
                    Cancel
                  </button>
                </div>
              </td>
              <td>
                FAIL
              </td>
              <td
                style="width: 500px;"
              >
                
                
                <span>
                  The instructor cannot be added for some reason
                </span>
              </td>
            </tr>
          </tbody>
        </table>
        <button
          class="btn btn-primary"
        >
          Add All Instructors
        </button>
      </div>
    </div>
  </div>
</tm-admin-home-page>
`;
